report/src/main/java/com/hartwig/actin/report/pdf/tables/treatment/ActinTrialGeneratorFunctions.java

```diff
@@ -0,0 +1,55 @@
+package com.hartwig.actin.report.pdf.tables.treatment;
+
+import static java.util.stream.Collectors.groupingBy;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import com.hartwig.actin.report.interpretation.EvaluatedCohort;
+import com.hartwig.actin.report.interpretation.EvaluatedCohortComparator;
+import com.hartwig.actin.report.pdf.util.Cells;
+import com.hartwig.actin.report.pdf.util.Styles;
+import com.itextpdf.layout.element.Paragraph;
+import com.itextpdf.layout.element.Table;
+import com.itextpdf.layout.element.Text;
+
+public final class ActinTrialGeneratorFunctions {
+
+    public static Stream<List<EvaluatedCohort>> streamSortedCohorts(List<EvaluatedCohort> cohorts) {
+        cohorts.sort(new EvaluatedCohortComparator());
+        Map<String, List<EvaluatedCohort>> cohortsByTrialId = cohorts.stream().collect(groupingBy(EvaluatedCohort::trialId));
+
+        return cohorts.stream().map(EvaluatedCohort::trialId).distinct().map(cohortsByTrialId::get);
+    }
+
+    public static String createCohortString(EvaluatedCohort cohort) {
+        String cohortText = cohort.cohort() == null ? "" : cohort.cohort();
+        if (cohort.isOpen() && !cohort.hasSlotsAvailable()) {
+            cohortText += " *";
+        }
+        return cohortText;
+    }
+
+    public static void addContentStreamToTable(Stream<String> cellContent, boolean deemphasizeContent, Table table) {
+        cellContent.map(text -> {
+            if (deemphasizeContent) {
+                return Cells.createContentNoBorderDeemphasize(text);
+            } else {
+                return Cells.createContentNoBorder(text);
+            }
+        }).forEach(table::addCell);
+    }
+
+    public static void insertTrialRow(List<EvaluatedCohort> cohortList, Table table, Table trialSubTable) {
+        EvaluatedCohort cohort = cohortList.get(0);
+        if (cohort != null) {
```

Comment:
I have not checked this exhaustively, but this looks sketchy upon first glance. Can't cohortList be empty? And why would the list itself contain a null value?